sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutor.java

```diff
@@ -943,4 +942,24 @@ public boolean onEmitFailure(SignalType signalType, Sinks.EmitResult emitResult)
             return false;
         }
     }
+
+    private static class SerializedCompleteEmitFailureHandler implements Sinks.EmitFailureHandler {
+
+        @Override
+        public boolean onEmitFailure(SignalType signalType, Sinks.EmitResult emitResult) {
+            if (emitResult.equals(Sinks.EmitResult.FAIL_NON_SERIALIZED)) {
+                logger.debug("SerializedCompleteEmitFailureHandler.onEmitFailure - Signal:{}, Result: {}", signalType, emitResult);
+
+                return true;
+            }
+
+            if (emitResult == Sinks.EmitResult.FAIL_CANCELLED || emitResult == Sinks.EmitResult.FAIL_TERMINATED) {
+                logger.debug("SerializedCompleteEmitFailureHandler.onEmitFailure - Main sink already completed, Signal:{}, Result: {}", signalType, emitResult);
```

Comment:
info or warning?

Reply:
No - Cancelled or Terminated is ok - just the "looser" on synchronization